src/app/models/grid.model.ts

```diff
@@ -1,26 +1,38 @@
-import { Cell } from "./cell.model";
+import { Cell } from './cell.model';
 
 export class Grid {
-    private rowCount: number;
-    private columnCount: number;
-    private cells: Cell[][];
+  private rowCount: number;
+  private columnCount: number;
+  private cells: Cell[][];
 
-    constructor(rows: number, columns: number) {
-        this.rowCount = rows;
-        this.columnCount = columns;
-        this.cells = new Array(this.rowCount);
-        for(let r = 0; r < this.rowCount; r++) {
-            this.cells[r] = new Array(this.rowCount);
-            for(let c = 0; c < this.columnCount; c++) {
-                this.cells[r][c] = new Cell(this, r, c);
-            }
-        }
+  constructor(rows: number, columns: number) {
+    this.rowCount = rows;
+    this.columnCount = columns;
+    this.cells = new Array(this.rowCount);
+    for (let r = 0; r < this.rowCount; r++) {
+      this.cells[r] = new Array(this.rowCount);
+      for (let c = 0; c < this.columnCount; c++) {
+        this.cells[r][c] = new Cell(this, r, c);
+      }
     }
+  }
 
-    getCell(row: number, col: number): Cell{
-        const normalizedRow = (row + this.rowCount) % this.rowCount;
-        const normalizedCol = (col + this.columnCount) % this.columnCount;
+  getCell(row: number, col: number): Cell {
+    const normalizedRow = (row + this.rowCount) % this.rowCount;
+    const normalizedCol = (col + this.columnCount) % this.columnCount;
 
-        return this.cells[normalizedRow][normalizedCol];
-    }
+    return this.cells[normalizedRow][normalizedCol];
+  }
+
+  getRandomCell() {
+    return this.getCell(this.generateRandomRow(), this.generateRandomColumn());
+  }
+
+  private generateRandomRow() {
```

Comment:
Here we have the same logic repeated 2 times. I'd propose to create a `RandomNumberHelper` and extract into a function in it. This way we won't need those 2 methods and will just have to call the method 2 times with `this.rowCount - 1` and `this.columnCount - 1` as parameter